llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -128,7 +128,7 @@ def _register_model_with_run_id_multiprocess(
         logging.getLogger('composer').setLevel(composer_logging_level)
 
     # Register model.
-    mlflow_logger.register_model_with_run_id(
+    mlflow_logger.register_model(
```

Comment:
we call this API because otherwise the run id doesn't get properly attached to the registered model, not because of retries.